### PR TITLE
add oauth and split states

### PR DESCRIPTION
**Pull request by RoriCremer**
Adds Google Oauth (Stolen from Terra for now)
Cleans up some of the splitting out of the logged in/out states
Adds better mocks to testing

### CODE REVIEW

File: src/components/HomeView.jsx [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-/* istanbul ignore next */
-function mapStateToProps(state) {
-  return {
-    datasets: state.datasets,
-    studies: state.studies,
-  };
-}
-
-export default connect(mapStateToProps)(HomeView);
+export default HomeView;
```
Comment (thathert) [inline src/components/HomeView.jsx:22]: I thought we wanted to connect this to the store?

File: src/components/WelcomeView.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+              tellus mollis orci, sed rhoncus pronin sapien nunc accuan eget.
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default WelcomeView;
```
Comment (thathert) [inline src/components/WelcomeView.jsx:66]: Connect?

File: src/index.jsx [hunk 3]
```diff
@@ -4,13 +4,11 @@ import ReactDOM from 'react-dom';
 
 import { Provider } from 'react-redux';
 import { AppContainer } from 'react-hot-loader';
-import { PersistGate } from 'redux-persist/lib/integration/react';
```
Comment (thathert) [inline src/index.jsx:7]: Can you remove redux-persist from package.json (and the package-lock.json)?

File: src/routes/Home.jsx [hunk 4]
```diff
@@ -21,64 +15,18 @@ const HomeContainer = styled(Container)`
...
-            size="xl"
-            textTransform="uppercase"
-            data-testid="Login"
-          >
-            <Icon name="sign-in" />
-            <Text ml={2}>Start</Text>
-          </Button>
+      <Background key="home" data-testid="home">
+        <HomeContainer>
+          <WelcomeView dispatch={dispatch} />
```
Comment (thathert) [inline src/routes/Home.jsx:29]: If you connect the WelcomeView component then you don't need to pass dispatch around like this.

**Review by thathert — CHANGES_REQUESTED** [review]
Looks good, have a couple changes.

File: src/routes/Home.jsx (hunk 4 above)
Comment (RoriCremer) [reply src/routes/Home.jsx:29]: let's circle back and talk about how we want to organize/connect this stuff

**Review by thathert — APPROVED** [review]
Looks good, merging.

**Pull request merged**